### ISSUE TITLE
[SR-15255] Windows: Calling URL(contentsOf:) repeatedly crashes

### ISSUE THREAD

**Issue by adam-fowler** [apple/swift-corelibs-foundation#4582]
|                  |                 |
   |------------------|-----------------|
   |Previous ID       | SR-15255      |
   |Radar             | None         |
   |Original Reporter | @adam-fowler      |
   |Type              | Bug    |

<details>
  <summary>Environment</summary>

Run inside GitHub action using \`compnerd/gha-setup-swift\`

``` java
jobs:
  windows:
    runs-on: windows-latest
    steps:
      - uses: compnerd/gha-setup-swift@main
        with:
          branch: swift-5.5-release
          tag: 5.5-RELEASE
      - uses: actions/checkout@v2
      - run: swift test

```

I don't have a Windows setup so cannot test locally

</details>

   <details>
  <summary>Additional Detail from JIRA</summary>

   |                  |                 |
   |------------------|-----------------|
   |Votes             | 0         |
   |Component/s       |     |
   |Labels            | Bug        |
   |Assignee          | None      |
   |Priority          | Medium      |

   

   md5: fad9bd9ca6233534ee2375c19875deab

  </details>

**Issue Description:**

The following code crashes on Windows.

``` java
func testURLs() throws {
    for _ in 0..<20 {
        if let url = URL(string: "https://raw.githubusercontent.com/jmespath/jmespath.test/master/tests/filters.json") {
            _ = try Data(contentsOf: url)
        }
    }
}

```

**Comment by stefanspringer1** [comment]
At least with Swift 6.1 it seems clear that only file URLs can be used in `Data(contentsOf:)` on Windows. The code above still crashes without a nice error message, but the following code

```swift
import Foundation

let url = URL(string: "https://raw.githubusercontent.com/jmespath/jmespath.test/master/tests/filters.json")!
do {
    _ = try Data(contentsOf: url)
} catch {
    print(error.localizedDescription)
}
```

prints the nicer info "The operation could not be completed. The specified URL type isn’t supported." which I suppose stems from the following code in the Foundation sources:

```swift
#if !FOUNDATION_FRAMEWORK
@_spi(SwiftCorelibsFoundation)
public dynamic init(_contentsOfRemote url: URL, options: ReadingOptions = []) throws {
    assert(!url.isFileURL)
    throw CocoaError(.fileReadUnsupportedScheme)
}
#endif
```

`Data(contentsOf:)` should be able to read from network URLs also on Windows, since it comes unexpected that this feature exist on macOS, but not on Windows.

**Comment by parkera** [comment]
Importing `FoundationNetworking` should be sufficient to make this work. See https://github.com/swiftlang/swift-corelibs-foundation/blob/cac38ff51cd4a120387afb02f065c3f86fcfd42a/Sources/FoundationNetworking/URLSession/NetworkingSpecific.swift#L40

**Comment by stefanspringer1** [comment]
No, importing FoundationNetworking suffices on Linux, but does not suffice on Windows (tested with Swift 6.1.1).

**Comment by parkera** [comment]
Is there some difference in the windows runtime with how dynamic replacement works? cc @compnerd

**Comment by compnerd** [comment]
@parkera no, not that I can think of at least.
